src/jobsautoreport/query.py

```diff
@@ -110,7 +110,11 @@ def _parse_step_events(
 
     @staticmethod
     def _parse_job(elastic_search_job: dict[Any, Any]) -> JobDetails:
-        return JobDetails.parse_obj(elastic_search_job)
+        parsed_job = JobDetails.parse_obj(elastic_search_job)
+        parsed_job.name = (
+            f"{parsed_job.refs.repo}-{parsed_job.refs.base_ref}-{parsed_job.context}"
+        )
+        return parsed_job
```

Comment:
don't modify the job name, keep JobDeteils as it is. Instead, create an object from the JobDetails that will be passed around to identify a job, something like that:

```
JobIdentifier:
    name
    context
    base_ref
    ...

    def pretty_name(self):
        return f"{self.repo}-{self.base_ref}-{self.context}"
```